pkg/test/framework/components/echo/options.go

```diff
@@ -0,0 +1,60 @@
+//  Copyright Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package echo
+
+import (
+	"fmt"
+
+	"istio.io/istio/pkg/test/framework/resource"
+)
+
+type ConfigOptions struct {
```

Comment:
You want to remove `With` from the Builder interface? That'd be a pretty big breaking change to cleanup but I can see the advantage for encouraging test authors to handle multicluster cases. 

I think I'd also suggest having `ForSingleCluster` return a ClusterContext to make the refactor a bit cleaner, rather than having `ForClusters(ctx.Clusters[0])` all over.

Reply:
Yeah that was my thought ... just make the breaking change now and switch everything over so that the implications for multicluster are more clear.

I suppose we could make the Builder implement `With` as well (i.e. it *is* a `ClusterContext`), but then we have to think about what the default set of clusters should be ... cluster[0]? all clusters? Ideally it should be all clusters, but I think it would be better to switch to the all clusters model now and make the single cluster tests do something special, like your `ForSingleCluster` example.

WDYT?

Reply:
For the initial PR I'd like to leave the top-level `With` as-is; there are a ton of code changes in similar areas here https://github.com/istio/istio/pull/24549 and merging would be a bit of a pain.

Reply:
I'll give it a go and revert if the merge is nuts.